tests/host_tools/cargo_build.py

```diff
@@ -55,24 +55,29 @@ def get_firecracker_binaries():
     Returns the location of the firecracker related binaries eventually after
     building them in case they do not exist at the specified root_path.
     """
+    if not hasattr(get_firecracker_binaries, "was_built"):
+        get_firecracker_binaries.was_built = False
+
     target = DEFAULT_BUILD_TARGET
     cd_cmd = "cd {}".format(FC_WORKSPACE_DIR)
     flags = 'RUSTFLAGS="{}"'.format(get_rustflags())
     cargo_cmd = "cargo build --release --target {}".format(target)
     cmd = "{} && {} {}".format(cd_cmd, flags, cargo_cmd)
 
-    utils.run_cmd(cmd)
-
     out_dir = "{target_dir}/{target}/release".format(
         target_dir=FC_WORKSPACE_TARGET_DIR, target=target
     )
     fc_bin_path = "{}/{}".format(out_dir, FC_BINARY_NAME)
     jailer_bin_path = "{}/{}".format(out_dir, JAILER_BINARY_NAME)
 
-    utils.run_cmd(
-        "strip --strip-debug {} {}"
-        .format(fc_bin_path, jailer_bin_path)
-    )
+    if not get_firecracker_binaries.was_built:
+        utils.run_cmd(cmd)
+
+        utils.run_cmd(
+            "strip --strip-debug {} {}"
+            .format(fc_bin_path, jailer_bin_path)
+        )
```

Comment:
Why do we `strip` binaries used in tests anyways? This step adds unnecessary time to tests.

@firecracker-microvm/compute-capsule I say we remove it.

Reply:
If we want our release binaries to be stripped, we also need to run the CI with them in this form.

Reply:
In theory, yes.
But we could also make the simplifying assumption that the `strip` binary in the container that we control is correct and does not corrupt the target binary.

This `strip` step/command adds 2 seconds to each `pytest` run (this cost is felt most during test development when we do `pytest` over and over with a single test), so it's not that bad. I would like to not pay those 2 seconds, but at the same time it's not such a big price to pay if we don't trust `strip` and want to verify.